eq-author/src/App/page/Logic/index.js

```diff
@@ -67,36 +69,80 @@ const LogicLink = styled(NavLink)`
   }
 `;
 
+const Badge = styled.span`
+  border-radius: 0.7em;
+  border: 1px solid ${colors.white};
+  background-color: ${colors.red};
+  color: white;
+  padding: 0.1em 0.3em;
+  font-weight: normal;
+  z-index: 2;
+  margin-left: auto;
+  line-height: 1;
+  font-size: 0.9rem;
+  pointer-events: none;
+  width: 20px;
+  height: 20px;
+`;
+
 export class UnwrappedLogicPage extends React.Component {
   static propTypes = {
     children: PropTypes.node.isRequired,
-    page: CustomPropTypes.page,
-    match: PropTypes.shape({
-      params: PropTypes.shape({
-        questionnaireId: PropTypes.string.isRequired,
-      }).isRequired,
-    }).isRequired,
+    data: PropTypes.shape({
+      page: CustomPropTypes.page,
+    }),
+    loading: PropTypes.bool.isRequired,
+    error: PropTypes.object, // eslint-disable-line
   };
 
   renderContent() {
-    const { children } = this.props;
+    const { children, data, loading, error } = this.props;
+    const page = get(data, "page", null);
+
+    if (loading) {
+      return <Loading height="20em">Loading routing</Loading>;
+    }
+
+    if (error || !page) {
+      return <Error>Something went wrong</Error>;
+    }
+
+    const TABS = [
+      {
+        key: `routing`,
+        label: "Routing logic",
+      },
+      {
+        key: `skip`,
+        label: "Skip logic",
+      },
+    ];
```

Comment:
By getting the data on line 100 this way, everytime you change the tab it completely blanks and shows the loading component for a split second before reloading

Reply:
Also the TABS are static so I think they might be able to be taken out of the rendeContent method